lib/teleterm/gateway/kube_cert_reissuer.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gateway
+
+import (
+	"context"
+	"crypto/tls"
+	"sync/atomic"
+
+	"github.com/gravitational/trace"
+)
+
+// kubeCertReissuer implements a simple single-kube cert reissuer that can be
+// used for kube local proxy middleware.
+type kubeCertReissuer struct {
+	cert          atomic.Value
+	onExpiredCert func(context.Context) error
+}
+
+func newKubeCertReissuer(cert tls.Certificate, onExpiredCert func(context.Context) error) *kubeCertReissuer {
```

Comment:
It is a bit unclear for me why is the difference between `kubeCertReissuer`and [*GatewayCertReissuer) ReissueCert](https://github.com/gravitational/teleport/blob/3ac5a0aedba7c50cb0eebf51a812d9dfec481d65/lib/teleterm/daemon/gateway_cert_reissuer.go#L81)

Reply:
gateway and cluster are injected along the way when GatewayCertReissuer callback is called. so at `kubeCertReissuer` level just trying to hide those. I am open to any suggestions though. I do feel bad when tracing this roundtrip.